### ISSUE TITLE
Feature Request: Removal of entries from RStudio history mechanism

### ISSUE THREAD

**Issue by PrivacyMatter** [rstudio/rstudio#4305]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.2.1153
    OS Version      : MacOS Mojave
    R Version       : 

### Describe the behavior you expected

(The feature request is based on this post that I wrote (and follow up responses) in RStudio Community: https://community.rstudio.com/t/removing-an-entry-from-rstudio-history/23324?u=privacy) 

I really like the RStudio history management mechanism, that goes way beyond what is provided by the simple R.history file. I think that adding a feature that enables removal of certain entries from the history, e.g. by choosing them in the history window and having some delete button (similar to removal of objects from environments).  I _think_ that it would also affect the history-based suggestions at the command line, which would be great, but even if it wouldn't, not seeing the expression when searching through history would be really useful.

In case that justification is needed, or would help:

The interpreter work style that R provides allows exploring data in many different ways, using incremental R statements or even code blocks from the command line.  Speaking for myself, even after working with R for a few years now(and even wrote some small ad-hoc libraries for self use), I am still making many mistakes, or just making many attempts to get the data I need.  I find History extremely useful to find expressions that I did find useful but they do not yet belong or ripe enough to be put in a a Rscript file.  The fact, though, that each such statement is surrounded by 5 attempts to get it right, makes it a bit harder to identify the one(s) ending up being what I was looking for.  

The mechanism is also extremely useful when you lose some temporary, in memory data  when RStudio crashes before saving .RData (here I'm referring to actual data frames or lists that are intermediate results of data processing in R).  The history mechanism greatly helps rebuilding that state, but again, it would be great if I would have the option to remove all the false attempts right after executing them and rebuilt my state much faster.

Bottom line, I think that adding more control on managing the history mechanism would be wonderful, and make it even more useful than it is today.

Thanks much.

**Comment by ronblum** [comment]
@PrivacyMatter Thanks for the feature suggestion! We'll take a look at it as we continue to develop new versions of RStudio.
